contribute/design/1.x/cstor/pool/pool_expansion_by_expanding_disk.md

```diff
@@ -0,0 +1,135 @@
+
+- CStor-pool Manger inbuilt CSPI controller watches for CSPI resources and tries to get the desired state via every sync operation. As part of reconciliation logic, it will iterate over all the BlockDevices that exist on CStorPoolInstance resource if it found BlockDevice resource capacity(considerable changes) is greater than the existing capacity of BlockDevice on CstorPoolInstanceBlockDevice then we will follow below-mentioned steps to perform the [pool expansion](#Pool_Expansion_Steps).
+- [What if cStor is consuming part of partitioned disk/entire partitioned disk? The User has to extend the partition manually after that NDM will update the capacity on the corresponding blockdevice CR. Once NDM updates the capacity on BlockDevice in next reconciliation of CSPI Pool Expansion will be triggered].
+- [What if only one blockdevice got expanded in case of raid/mirror configuration? Pool expansion operation will be triggered but there won't be any change in the size of the pool].
+
+Currently CstorPoolInstanceBlockDevice holds below configurations 
+```go
+// CStorPoolInstanceBlockDevice contains the details of block devices that
+// constitutes a raid group.
+type CStorPoolInstanceBlockDevice struct {
+	// BlockDeviceName is the name of the block device.
+	BlockDeviceName string `json:"blockDeviceName"`
+
+	// Capacity is the capacity of the block device.
+	// It is system generated
+	Capacity string `json:"capacity"`
+
+	// DevLink is the dev link for block devices
+	DevLink string `json:"devLink"`
+}
+```
+
+### Steps to perform user stories
+
+- User has to make sure that blockdevice capacity to needs to be updated.
+
+#### Pool Expansion Steps:
+
+Step1: Remove the buffering partition if exists which was created by cstor-pool container(ex: sdb9).
+
+Step2: Expand the existing partition by executing following comamnd
+       ```sh
+       parted </disk> resizepart <partition_number> 100%
+       ```
+
+Step3: Inform the pool saying disk was expanded by triggering following command
+       ```sh
+       zpool online -e <pool_name> <disk_name>
+       ```
+Step4: If above steps are successfull then `Capacity` field in CStorPoolInstanceBlockDevice will be updated with latest size.
+NOTE: Above steps automates the steps mentioned in [doc](https://github.com/openebs/openebs-docs/blob/day_2_ops/docs/resize-single-disk-pool.md).
+
+### Low Level Design
+
+#### Work Flow
+
+- User/Admin will make sure the expanded size should reflect on corresponding blockdevice CR.
+- During CSPI reconciliation cstor-pool-mgmt verifies whether the disk was expanded or not by comparing the capacity of CStorPoolInstanceBlockDevice with BlockDevice if there is a considerable change in the capacity then cstor-pool-mgmt will trigger pool expansion steps.
+
+NOTE: The Pool will be expanded only if all the blockdevices in raidgroup were expanded in case of mirror/raid configuration. If it is stripe configuration expanding one blockdevice will expand the pool.
```

Comment:
Does this line and the below statement contradict?
```
[What if only one blockdevice got expanded in case of raid/mirror configuration? Pool expansion operation will be triggered but there won't be any change in the size of the pool]
```

Reply:
No both are the same, comment conveying there won't be any increase in the size of the pool if only one blockdevice is expanded in mirror or raidz configuration. The Note is conveying expand of pool will happen only when all the block devices in that raidgroup are expanded.